### PR TITLE
PLANNER-2462 Update react-scripts

### PR DESCRIPTION
**Pull request by yurloc**
### Dependencies updates
- `react-scripts`, `react` and `redux` to latest.
- `react-router` to 5.x latest (6.x has a long [upgrade recipe](https://reactrouter.com/en/v6.3.0/upgrading/v5)).
- PatternFly to latest.
- `leaflet-geosearch` to latest.
  - Is now written in TypeScript, which allowed me to remove the custom typings file (`leaflet-geosearch.d.ts`).

### Dev dependencies updates
- Dropped Enzyme (seems dead) and replaced it with [React Testing Library](https://create-react-app.dev/docs/running-tests/#react-testing-library).
- Updated react, redux, router and jest types.
- Updated ESLint config dependencies and removed plugin dependencies overrides.
- Updated `typescript` from 3.8.3 to 4.1 (latest is 4.9 now).

### Source code changes
- [fix] Search results now have unique [keys](https://reactjs.org/docs/lists-and-keys.html#keys).
- Connection error dialog updates:
  - Removed the close button.
  - Replaced title icon in favor of code simplification.
  - Improved the error message.
- Load demo is now a primary button (for consistency with other buttons).

### Known issues
- Background image is not visible.
- Several disabled tests.

### JIRA
https://issues.redhat.com/browse/PLANNER-2462

Also updates PaternFly to latest and resolves https://issues.redhat.com/browse/PLANNER-2031.

### Referenced pull requests
- Closes #481
- Closes #482
- Closes #484
- Closes #486
- Closes #742
- Closes #743
- Closes #755
- Closes #764
- Closes #777
- Closes #778
- Closes #779
- Closes #797

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  please add comment: <b>Jenkins (re)run [optaweb-vehicle-routing] tests</b>
  
- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job:  
    please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job:  
    add the label `run_fdb`

- for a <b>compile downstream build</b>  
  please add comment: <b>Jenkins run cdb</b>

- for a <b>full production downstream build</b>  
  please add comment: <b>Jenkins execute product fdb</b>

- for an <b>upstream build</b>  
  please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [optaweb-vehicle-routing] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [optaweb-vehicle-routing] quarkus-branch</b>
</details>

### CI Status

You can check OptaPlanner repositories CI status from [Chain Status webpage](https://kiegroup.github.io/optaplanner/).

### CODE REVIEW

File: optaweb-vehicle-routing-frontend/src/store/client/reducers.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export const initialViewportState: UserViewport = {
   zoom: 2,
 };
 
+// eslint-disable-next-line @typescript-eslint/default-param-last
```
Comment (Christopher-Chianelli) [inline optaweb-vehicle-routing-frontend/src/store/client/reducers.ts:9]: I notice this eslint disable on several lines in the PR. Should the rule be disabled in .eslintrc?

**Review by Christopher-Chianelli — APPROVED** [review]
Changes look good to me; tested locally. Note: it appears the app is incompatible with older databases for the map data; I needed to remove the old data in optaweb-vehicle-routing for the app to run. I don't know if you want to fix disabled tests in this PR or in another PR

**Comment by yurloc** [comment]
Do not merge.

File: optaweb-vehicle-routing-frontend/src/store/client/reducers.ts (hunk 1 above)
Comment (yurloc) [reply optaweb-vehicle-routing-frontend/src/store/client/reducers.ts:9]: I think the rule makes sense in general. It's just incompatible with the reducer signature:

From https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers:

> Reducers are functions that take the current state and an action as arguments, and return a new state result. In other words, `(state, action) => newState`.

and

> Reducers normally use ES6 default argument syntax to provide initial state: `(state = initialState, action)`.

**Pull request merged**